stream_alert/alert_processor/outputs.py

```diff
@@ -199,6 +196,245 @@ def dispatch(self, **kwargs):
+        ])
+
+    @staticmethod
+    def _get_endpoint(base_url, endpoint):
+        """Helper to get the full url for a PagerDuty Incidents endpoint.
+
+        Args:
+            base_url (str): Base URL for the API
+            endpoint (str): Endpoint that we want the full URL for
+
+        Returns:
+            str: Full URL of the provided endpoint
+        """
+        return os.path.join(base_url, endpoint)
+
+    def _check_exists_get_id(self, filter_str, url, target_key):
+        """Generic method to run a search in the PagerDuty REST API and return the id
+        of the first occurence from the results.
+
+        Args:
+            filter_str (str): The query filter to search for in the API
+            url (str): The url to send the requests to in the API
+            target_key (str): The key to extract in the returned results
+
+        Returns:
+            str: ID of the targeted element that matches the provided filter or
+                 False if a matching element does not exists.
+        """
+        params = {
+            'query': '"{}"'.format(filter_str)
+        }
+        resp = self._get_request(url, params, self.headers, False)
+        if not self._check_http_response(resp):
+            return False
+
+        response = resp.json()
+
+        # If there are results, get the first occurence from the list
+        target = response.get(target_key, False)
+        if response and target:
+            return target[0]['id']
+
+        return False
+
+    def _user_verify(self, user):
+        """Method to verify the existance of an user with the API
+
+        Args:
+            user (str): User to query about in the API.
+
+        Returns:
+            dict or False: JSON object be used in the API call, containing the user_id
+                           and user_reference. False if user is not found
+        """
+        users_url = self._get_endpoint(self.base_url, self.USERS_ENDPOINT)
+
+        return self._item_verify(users_url, user, self.USERS_ENDPOINT, 'user_reference')
+
+    def _policy_verify(self, policy):
+        """Method to verify the existance of a escalation policy with the API
+
+        Args:
+            api_url (str): Base URL of the API
+            policy (str): Escalation policy to query about in the API
+            headers (dict): Headers used for API authentication
+
+        Returns:
+            dict: JSON object be used in the API call, containing the policy_id
+                  and escalation_policy_reference
+        """
+        policies_url = self._get_endpoint(self.base_url, self.POLICIES_ENDPOINT)
+
+        return self._item_verify(policies_url, policy, self.POLICIES_ENDPOINT,
+                                 'escalation_policy_reference')
+
+    def _service_verify(self, service):
+        """Method to verify the existance of a service with the API
+
+        Args:
+            api_url (str): Base URL of the API
```

Comment:
I think this args list needs updated to only include `service`

Reply:
@javuto still seeing `api_url` in this list but this isn't an arg in the function declaration 🤔